File: client/clojure/src/cdisc_library_api/specs/cdash_class_field.clj
(ns cdisc-library-api.specs.cdash-class-field
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.cdash-class-field-links :refer :all]
            )
  (:import (java.io File)))


(def cdash-class-field-data
  {
   (ds/opt :ordinal) string?
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :definition) string?
   (ds/opt :questionText) string?
   (ds/opt :prompt) string?
   (ds/opt :simpleDatatype) string?
   (ds/opt :implementationNotes) string?
   (ds/opt :mappingInstructions) string?
   (ds/opt :_links) cdash-class-field-links-spec
   })

(def cdash-class-field-spec
  (ds/spec
    {:name ::cdash-class-field
     :spec cdash-class-field-data}))

File: client/scalaz/src/main/scala/org/openapitools/client/api/CdashClassField.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CdashClassField._

case class CdashClassField (
  ordinal: Option[String],
name: Option[String],
label: Option[String],
definition: Option[String],
questionText: Option[String],
prompt: Option[String],
simpleDatatype: Option[String],
implementationNotes: Option[String],
mappingInstructions: Option[String],
links: Option[CdashClassFieldLinks])

object CdashClassField {
  import DateTimeCodecs._

  implicit val CdashClassFieldCodecJson: CodecJson[CdashClassField] = CodecJson.derive[CdashClassField]
  implicit val CdashClassFieldDecoder: EntityDecoder[CdashClassField] = jsonOf[CdashClassField]
  implicit val CdashClassFieldEncoder: EntityEncoder[CdashClassField] = jsonEncoderOf[CdashClassField]
}
